usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js

```diff
@@ -0,0 +1,832 @@
+            }
+            var newLine = polyLine ? false : true;
+            if (newLine) { 
+                print("BEGINNING NEW LINE______________");
+            }
+            var lineProperties = DEFAULT_LINE_PROPERTIES;
+            var linePointsCount;
+            var paintSphereDimensions = Entities.getEntityProperties(_this.entityID, 'dimensions').dimensions;
+            currentStrokeWidth = paintSphereDimensions.x;
+            if (!newLine) { // maybe editing existing line
+                var previousLineProperties = Entities.getEntityProperties(polyLine, ['linePoints', 'normals', 
+                    'strokeWidths', 'age']);
+                linePointsCount = previousLineProperties.linePoints.length;
+                if (linePointsCount > MAX_LINE_POINTS) { // too many line points, start new line connected to previous point
+                    // print("CONTINUING LINE ************************");
+                    newLine = true;
+                    if (onBoard) {
+                        lineStartPosition = Vec3.subtract(previousLinePoint, Vec3.multiply(previousNormal, 
+                            STROKE_FORWARD_OFFSET_M));
+                    } else {
+                        lineStartPosition = previousLinePoint;
+                    }
+                    displacementFromStart = Vec3.subtract(currentLinePoint, lineStartPosition);
+                    displacementFromStart = Vec3.subtract(displacementFromStart, Vec3.multiply(normal, 
+                        STROKE_FORWARD_OFFSET_M));
+                    // print("DISPLACEMENT FROM NEW START ", JSON.stringify(displacementFromStart));
+                } else { // actually editing the previous line
+                    if (injector) {
+                        injector.options = {
+                            position: currentLinePoint,
+                            volume: DRAW_SOUND_VOLUME
+                        };
+                    }
+                    lineProperties.linePoints.push(displacementFromStart);
+                    lineProperties.normals.push(normal);
+                    lineProperties.strokeWidths.push(currentStrokeWidth);
+                    if (!onBoard) {
+                        lineProperties.lifetime = previousLineProperties.age + DECAY_TIME_S;
+                    }
+                    Entities.editEntity(polyLine, {
+                        linePoints: lineProperties.linePoints,
+                        normals: lineProperties.normals,
+                        strokeWidths: lineProperties.strokeWidths,
+                        lifetime: lineProperties.lifetime,
+                        faceCamera: onBoard ? false : true
+                    });
+                }
+            }
+            // new line due to just beginning to draw or starting new to continue line with too many points. 
+            // All lines have some previous data saved from the initial point, actual new lines have no line points yet
+            if (newLine) {
```

Comment:
can't this be an `else` above?

Reply:
no, if the current line has too many points, a new one is begun but that is determined in the statement above

Reply:
ahhh got it